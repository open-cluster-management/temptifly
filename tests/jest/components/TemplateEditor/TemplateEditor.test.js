'use strict'

import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import TemplateEditor from '../../../../src/TemplateEditor'
import { render, screen, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import userEvent from '@testing-library/user-event'
import template from '../../../../example/templates/template.hbs'
import controlData from '../../../../example/controlData/ControlData'

//window.MonacoEnvironment = {
//  getWorkerUrl: function() {
//    return `${config.contextPath}/editor.worker.js`
//  }
//}

export const portals = Object.freeze({
  cancelBtn: 'cancel-button-portal-id',
  createBtn: 'create-button-portal-id',
  editBtn: 'edit-button-portal-id'
})

const i18n = (key) => {
  return key
}

describe('TemplateEditor component',() => {
  it('renders as expected',  async () => {

    jest.setTimeout(30000)

    window.scrollBy = () => {}
    const createResource = (json) => {
      expect(json).toMatchSnapshot()
    }
    const createControl={
      createResource,
    }
    const Component = () => {
      return (
        <BrowserRouter>
          <div id={portals.createBtn} />
          <TemplateEditor
            template={template}
            controlData={controlData}
            createControl={createControl}
            portals={portals}
            type={'application'}
            title={'creation.app.yaml'}
            i18n={i18n}
            initialOpen={true}
          />
        </BrowserRouter>
      )
    }
    const { getByTestId, asFragment, getByRole, getByText, container } = render(<Component />)

    // text
    userEvent.type(getByTestId('text-eman'), 'test')
    // card
    userEvent.click(getByTestId('card-cluster.create.baremetal.subtitle'))

    // combobox
    await waitFor(() => expect(screen.getByTestId('combo-imageSet')).toBeInTheDocument())
    userEvent.type(getByTestId('combo-imageSet'), '4.3.40-x86_64')
    // tree select
    userEvent.type(getByTestId('tree-workerType'), 'highmem-16')
    // table
    userEvent.click(container.querySelector('[name="check-all"]'))
    userEvent.click(getByRole('button', { name: /edit row 1/i }))
    userEvent.click(getByRole('button', { name: /save edits for row 1/i }))
    userEvent.click(getByRole('button', { name: /edit row 2/i }))
    userEvent.click(getByRole('button', { name: /cancel edits for row 2/i }))
    userEvent.click(getByText('creation.ocp.host.name'))
    userEvent.click(getByText('creation.ocp.host.name'))
    userEvent.type(getByRole('textbox', { name: /search input/i }), 'bma1')
    expect(asFragment()).toMatchSnapshot()

    // create json
    userEvent.click(getByTestId('create-button-portal-id'))
  })
})
