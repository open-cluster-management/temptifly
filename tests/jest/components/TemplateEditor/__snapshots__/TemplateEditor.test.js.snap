// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateEditor component renders as expected 1`] = `
<div
  className="temptifly"
>
  <div
    className="creation-view-split-container"
  >
    <div
      className="creation-view-controls-container"
    >
      <div
        className="creation-view-controls"
        onScroll={[Function]}
      >
        <div
          id="notifications"
        />
        <div
          className="content"
        >
          <div
            aria-label={undefined}
            className="creation-view-controls-title"
            id="main-"
            onClick={[Function]}
            onKeyPress={[Function]}
            role="button"
            tabIndex="0"
            title={undefined}
          >
            <div
              className="creation-view-controls-note"
            />
            <div
              className="creation-view-controls-title-normal-container"
            />
          </div>
          <div
            className="creation-view-controls-section"
          >
            <div
              className="creation-view-controls-textbox"
              style={
                Object {
                  "display": "",
                }
              }
            >
              <span
                style={
                  Object {
                    "alignItems": "baseline",
                    "display": "flex",
                  }
                }
              >
                <label
                  className="creation-view-controls-textbox-title"
                  htmlFor="name"
                />
              </span>
              <div
                className="bx--form-item"
              >
                <input
                  autoComplete="new-password"
                  className="bx--text-input bx--text__input"
                  disabled={false}
                  id="name"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder={undefined}
                  spellCheck={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="creation-view-controls-textbox"
              style={
                Object {
                  "display": "",
                }
              }
            >
              <span
                style={
                  Object {
                    "alignItems": "baseline",
                    "display": "flex",
                  }
                }
              >
                <label
                  className="creation-view-controls-textbox-title"
                  htmlFor="namespace"
                />
              </span>
              <div
                className="bx--form-item"
              >
                <input
                  autoComplete="new-password"
                  className="bx--text-input bx--text__input"
                  disabled={false}
                  id="namespace"
                  onChange={[Function]}
                  onClick={[Function]}
                  placeholder={undefined}
                  spellCheck={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="creation-view-controls-container-blurr bottom"
      />
    </div>
  </div>
</div>
`;
